### PR TITLE
Adds info on beta linters, re-orders the markup to match TOC layout

### PR DESCRIPTION
**Pull request by sardaukar**
- more info on Go linting
- adds info on TS and Elixir linting

### CODE REVIEW

**Comment by sardaukar** [comment]
Pinging @jakecraige and @jayroh for input on this, not 100% sure on the new linter config keys

**Comment by jayroh** [comment]
The new linters can remain off the radar for the time being while we put them through their paces with a small subset of customers and test repos. So no worries on those two in particular (for now). Once they're ready to go we'll get a PR for them up 😄

**Comment by croaky** [comment]
The "document Golint" part looks good to merge at your discretion!

File: app/views/pages/configuration.haml [hunk 1]
```diff
@@ -399,47 +452,80 @@
...
+          = link_to "Credo docs",
+            "http://credo-ci.org/",
+            new_window_options
+        and reference it in
+        %em.code-inline .hound.yml
+
+        %code.code-block
+          :preserve
+            elixir:
+              config_file: .credo.exs
```
Comment (jakecraige) [inline app/views/pages/configuration.haml]: The config file is good here, but you would also need to specify `enabled: true` for it to work while in beta.

**Comment by jakecraige** [comment]
Config stuff looks good for Elixir. Had one comment about making sure it's enabled. It's been on a few internal repos for the past few weeks without issue so I'd be comfortable adding it as a public beta

File: app/views/pages/configuration.haml [hunk 2]
```diff
@@ -23,6 +23,10 @@
         = link_to "ESLint (beta)", "#eslint", class: "docs-nav-link"
       %li
         = link_to "Python (beta)", "#python", class: "docs-nav-link"
+      %li
+        = link_to "TSLint (beta)", "#tslint", class: "docs-nav-link"
+      %li
+        = link_to "Elixir (beta)", "#elixir", class: "docs-nav-link"
```
Comment (teoljungberg) [inline app/views/pages/configuration.haml]: We haven't even added support for Elixir in Hound proper, which means we haven't even tested it out.

I'd personally prefer if we tested it out internally first before releasing it into the public.

Thoughts?

Comment (jakecraige) [inline app/views/pages/configuration.haml]: What do you mean? It is in Hound proper, https://github.com/houndci/hound/pull/1268. As briefly discussed in Slack we have been using it internally on a few projects with no issues so far, so I'd be okay with having this public beta. But not opposed to waiting a bit longer if we want. I don't predict any changes in that time though

**Comment by croaky** [comment]
I agree Elixir seems like it's been used a bunch on thoughtbot projects and we should feel comfortable releasing documentation on it now.

**Comment by croaky** [comment]
Do we need the "beta" distinction at all? Should "beta" for Hound mean simply undocumented and unpromoted on the website, but able to be enabled by thoughtbot or people following along on the OSS projects?

**Comment by gylaz** [comment]
I think we should rename "beta" linters to "disabled by default", since we'll need that now. For example, ESLint is no longer "beta" but it's disabled to favor JSHint out-of-the-box.

**Comment by croaky** [comment]
> rename "beta" linters to "disabled by default"

Nice. More actionable

File: app/views/pages/configuration.haml (hunk 2 above)
Comment (teoljungberg) [inline app/views/pages/configuration.haml]: > What do you mean? It is in Hound proper, #1268.

You're totally right, I missed out upon that.

If you're happy with how stable it is, I am too!

**Comment by gylaz** [comment]
Made the requested changes here, plus removed the "beta" meaning. Going to merge this.

**Pull request merged**